rfcs/0123-flake-names.md

```diff
@@ -0,0 +1,89 @@
+---
+feature: flake-names
+start-date: 2022-03-12
+author: Anselm Schüler
+co-authors: None
+shepherd-team: None
+shepherd-leader: None
+related-issues: None
+---
+
+# Summary
+[summary]: #summary
+
+Flakes can declare the field `name`.  
+It represents the name of the flake.  
+The derivations for a flake are no longer called `source`, but use the flake name.
+
+# Motivation
+[motivation]: #motivation
+
+Flake-centric workflows often end up with a lot of derivations named “source”, and it’s difficult to navigate this.
+This can also make flake-centric commands friendlier and easier to approach.
+
+# Detailed design
+[design]: #detailed-design
+
+A new supported property for flakes is introduced, `name`.  
+The derivation that contains the flake’s content is called `flake-source-${name}`  
+If a source control tag (e.g. git tag such as V3) is available, it is `flake-source-${name}-${tag}`.
```

Comment:
This isn’t really possible without breaking purity afaik: The same git commit can have or not a tag attached to it (since tags are mutable), and evaluating a flake from that commit with or without the tag would yield a different result

Reply:
I thought I had removed this section? Someone already raised this issue.
Yes, that's a good point.